torchio/data/image.py

```diff
@@ -309,11 +330,35 @@ def _load(self) -> Tuple[torch.Tensor, np.ndarray]:
         """
         if self._loaded:
             return
-        tensor, affine = read_image(self.path)
+
+        paths = self.path if isinstance(self.path, list) else [self.path]
+        tensor, affine = read_image(paths[0])
         tensor = self.parse_tensor_shape(tensor)
 
         if self.check_nans and torch.isnan(tensor).any():
-            warnings.warn(f'NaNs found in file "{self.path}"')
+            warnings.warn(f'NaNs found in file "{paths[0]}"')
+
+        tensors = [tensor]
+        for path in paths[1:]:
+            new_tensor, new_affine = read_image(path)
+            new_tensor = self.parse_tensor_shape(new_tensor)
+
+            if self.check_nans and torch.isnan(tensor).any():
+                warnings.warn(f'NaNs found in file "{path}"')
+
+            if not np.array_equal(affine, new_affine):
+                message = 'Files have different affine matrices'
+                warnings.warn(message, RuntimeWarning)
+
+            if not tensor.shape[1:] == new_tensor.shape[1:]:
+                message = f'Files shape do not match, found {tensor.shape}' \
+                          f'and {new_tensor.shape}'
+                RuntimeError(message)
+
+            tensors.append(new_tensor)
+
+        tensor = torch.cat(tensors, dim=0)
```

Comment:
I think you can omit `dim=0`.

Reply:
Maybe, but isn't it better if it's explicit?

Reply:
I don't think it helps in this case, as it is the default. It's like using `print(s, sep='\n')` instead of `print(s)`.